add sync and async test guide at contributing.md (#2694)
open-telemetry/opentelemetry-python-contrib: CONTRIBUTING.md

  - ex. <https://github.com/open-telemetry/opentelemetry-python-contrib/blob/2518a4ac07cb62ad6587dd8f6cbb5f8663a7e179/instrumentation/opentelemetry-instrumentation-requests/src/opentelemetry/instrumentation/requests/__init__.py#L234>
- Appropriate error handling
  - ex. <https://github.com/open-telemetry/opentelemetry-python-contrib/blob/2518a4ac07cb62ad6587dd8f6cbb5f8663a7e179/instrumentation/opentelemetry-instrumentation-requests/src/opentelemetry/instrumentation/requests/__init__.py#L220>
- Isolate sync and async test

>   - For synchronous tests, the typical test class structure is `class TestXxxxXxxx(TestBase)`. However, if you want to write asynchronous tests, you should create a new class that inherits from `IsolatedAsyncioTestCase` instead. Adding asynchronous tests to a common test class can lead to tests passing without actually running, which can be misleading.

Reviewer: ```suggestion
  - For synchronous tests, the typical test case class is inherited from `opentelemetry.test.test_base.TestBase`. However, if you want to write asynchronous tests, the test case class should inherits also from `IsolatedAsyncioTestCase`. Adding asynchronous tests to a common test class can lead to tests passing without actually running, which can be misleading.
```

Author: Thanks!
